utility/types/message_staking.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/pokt-network/pocket/shared/converters"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+)
+
+// This file captures basic logic common across all the actors that need to stake regardless of their responsibility.
+
+const (
+	httpsPrefix      = "https://"
+	httpPrefix       = "http://"
+	colon            = ":"
+	period           = "."
+	invalidURLPrefix = "the url must start with http:// or https://"
+	portRequired     = "a port is required"
+	nonNumberPort    = "invalid port, cant convert to integer"
+	portOutOfRange   = "invalid port, out of valid port range"
+	noPeriod         = "must contain one '.'"
+	maxPort          = 65535
```

Comment:
Not sure this file is the best spot for these. Some of them are quite general purpose